spec/acceptance/exec_powershell_spec.rb

```diff
@@ -322,6 +322,17 @@
     it_should_behave_like 'should fail', pexception, /We are writing an error/i
   end
 
+  describe 'should error if timeout is exceeded' do
+    ptimeoutexception = <<-MANIFEST
+      exec{'PowershellException':
+        command  => 'Write-Host "Going to sleep now..."; Start-Sleep 5',
+        timeout  => 2,
+        provider => powershell,
+      }
+    MANIFEST
+    it_should_behave_like 'should fail', ptimeoutexception
```

Comment:
1. Using this shared example, not I can't detect that
2. Yes you're right, the timeout errors always manifests as an Error.  Unlike https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider/exec.rb#L55 where it raises a `Timeout::Error` type